### PR TITLE
applications: asset_tracker_v2: Rework passive mode

### PR DESCRIPTION
**Pull request by simensrostad**
Rework passive mode:
- Use `passive mode` as the default mode for Thingy:91 builds.
- Disable wakeup mode and increase ODR for ADXL362 to increase
  its sensitivity.
- Update the logic in main.c to ensure that the device samples data as
  long as its considered active in passive mode.
- Remove "faking" of movement using button 2 on the DK.

### CODE REVIEW

**Comment by simensrostad** [comment]
Depends on https://github.com/zephyrproject-rtos/zephyr/pull/50818

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -126,6 +126,7 @@ nRF9160: Asset Tracker v2
 
   * The application now uses the new LwM2M location assistance objects through the :ref:`lib_lwm2m_location_assistance` library.
   * Added handling for the new data receive events in the :ref:`lib_nrf_cloud` library.
+  * Use passive mode as the default mode for Thingy:91 builds.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:129]:
```suggestion
  * The application now uses passive mode as the default mode for Thingy:91 builds.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -126,6 +126,7 @@ nRF9160: Asset Tracker v2
 
   * The application now uses the new LwM2M location assistance objects through the :ref:`lib_lwm2m_location_assistance` library.
   * Added handling for the new data receive events in the :ref:`lib_nrf_cloud` library.
+  * The application now uses passive mode as the default mode for Thingy:91 builds.
```
Comment (balaji-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:129]: dont we need to mention the change about the (in)activity thresholds too?

File: applications/asset_tracker_v2/boards/thingy91_nrf9160_ns.overlay [hunk 3]
```diff
@@ -20,7 +20,6 @@
 
 &spi3 {
 	adxl362: adxl362@0 {
-		wakeup-mode;
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/boards/thingy91_nrf9160_ns.overlay:23]: Why is this removed?

Comment (maxd-nordic) [reply applications/asset_tracker_v2/boards/thingy91_nrf9160_ns.overlay:23]: There is a problem with wake-up mode and setting the ODR (sampling rate) of the accelerometer.
This is to enable higher ODR I think.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (maxd-nordic) [reply doc/nrf/releases/release-notes-changelog.rst:129]: I think changing the default values of the thresholds is a minor change. Since they are so application-specific, customers will tweak them anyways.

**Review by balaji-nordic — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]
